porting-a-new-endpoint.md

```diff
@@ -0,0 +1,201 @@
+5. [Expose the Endpoint via FFI](#expose-the-endpoint-via-ffi): Update the Foreign Function Interface bindings so that the new endpoint can be accessed from Python
+6. [Implement the Service Logic](#implement-the-service-logic): Convert the coming proto struct and invoke the store.
+7. [Store Implementation](#store-implementation): Add a method to the Store interface to handle the new endpoint's database logic (implementation can be empty for now).
+8. [Write Tests](#write-tests): If required, add Go unit tests for isolated parts like validation, and rely on existing Python tests for integration.
+9. [Run Tests](#run-tests): Finally, run the Python integration tests to ensure that the Go implementation behaves consistently with the Python version. If required, adjust the tests for any differences in behavior between Go and Python.
+
+As mentioned elsewhere, the Go implementation of the tracking server is currently incomplete. This guide outlines how to implement a missing endpoint.
+
+## View Missing Endpoints
+
+Run `mage endpoints` to view which endpoints are not yet implemented.
+
+In this guide, we will implement `deleteTag`.
+
+```
++------------------------+------------------------------+-------------+
+|        SERVICE         |           ENDPOINT           | IMPLEMENTED |
++------------------------+------------------------------+-------------+
+...
++------------------------+------------------------------+-------------+
+| MlflowService          | deleteTag                    |     No      |
++------------------------+------------------------------+-------------+
+```
+
+## Enable the Endpoint
+
+Add the missing endpoint to [endpoints.go](magefiles/generate/endpoints.go) under the correct service. After that, run `mage generate`.
+
+This should update the following files:
+
+### Add a New Route
+
+In [pkg/server/routes/tracking.g.go](pkg/server/routes/tracking.g.go), add the following:
+
+```go
+	app.Post("/mlflow/runs/delete-tag", func(ctx *fiber.Ctx) error {
+		input := &protos.DeleteTag{}
+		if err := parser.ParseBody(ctx, input); err != nil {
+			return err
+		}
+		output, err := service.DeleteTag(utils.NewContextWithLoggerFromFiberContext(ctx), input)
+		if err != nil {
+			return err
+		}
+		return ctx.JSON(output)
+	})
+```
+
+This means that our Go server will now process the API request.
+
+### New Service Method
+
+In [pkg/contract/service/tracking.g.go](pkg/contract/service/tracking.g.go), the interface has a new function that we need to implement, which is used in the routes file above.
+
+```go
+type TrackingService interface {
+    ...
+	DeleteTag(ctx context.Context, input *protos.DeleteTag) (*protos.DeleteTag_Response, *contract.Error)
+}
+```
+
+### New FFI Endpoint
+
+In [pkg/lib/tracking.g.go](pkg/lib/tracking.g.go), we need to expose that same service method so it can be called directly from Python via FFI. More on that later.
+
```

Comment:
The `.g.go` suffix indicates a generated file, right? why are we editing it manually?